File: t-tasks/src/main/java/com/teo/ttasks/data/TaskListsAdapter.java
package com.teo.ttasks.data;

import android.content.Context;
import android.support.annotation.LayoutRes;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.TextView;

import com.teo.ttasks.R;
import com.teo.ttasks.data.model.TaskList;

public class TaskListsAdapter extends ArrayAdapter<TaskList> {

    private static int layoutResId = R.layout.spinner_item_task_list;
    private static int layoutResDropDownId = R.layout.spinner_item_task_list_dropdown;

    public TaskListsAdapter(Context context) {
        super(context, layoutResId);
    }

    @Override
    public void setDropDownViewResource(@LayoutRes int resource) {
        layoutResDropDownId = resource;
        super.setDropDownViewResource(resource);
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        // Get the data item for this position
        TaskList taskList = getItem(position);
        // Check if an existing view is being reused, otherwise inflate the view
        ViewHolder viewHolder; // view lookup cache stored in tag
        if (convertView == null) {
            viewHolder = new ViewHolder();
            LayoutInflater inflater = LayoutInflater.from(getContext());
            convertView = inflater.inflate(layoutResId, parent, false);
            viewHolder.name = (TextView) convertView.findViewById(R.id.taskListTitle);
            convertView.setTag(viewHolder);
        } else {
            viewHolder = (ViewHolder) convertView.getTag();
        }
        // Populate the data into the template view using the data object
        viewHolder.name.setText(taskList.getTitle());
        // Return the completed view to render on screen
        return convertView;
    }

    @Override
    public View getDropDownView(int position, View convertView, ViewGroup parent) {
        TaskList taskList = getItem(position);
        ViewHolder viewHolder;
        if (convertView == null) {
            viewHolder = new ViewHolder();
            LayoutInflater inflater = LayoutInflater.from(getContext());
            convertView = inflater.inflate(layoutResDropDownId, parent, false);
            viewHolder.name = (TextView) convertView.findViewById(R.id.taskListTitle);
            convertView.setTag(viewHolder);
        } else {
            viewHolder = (ViewHolder) convertView.getTag();
        }
        viewHolder.name.setText(taskList.getTitle());
        return convertView;
    }

    // View lookup cache
    private static class ViewHolder {
        TextView name;
    }
}

File: t-tasks/src/main/java/com/teo/ttasks/data/TaskListsAdapter.kt
package com.teo.ttasks.data

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import android.widget.TextView
import androidx.annotation.LayoutRes
import com.teo.ttasks.R
import com.teo.ttasks.data.model.TaskList

class TaskListsAdapter(
    context: Context,
    taskLists: List<TaskList> = emptyList()
) : ArrayAdapter<TaskList>(context, layoutResId, taskLists) {

    companion object {
        private const val layoutResId = R.layout.spinner_item_task_list
        private var layoutResDropDownId = R.layout.spinner_item_task_list_dropdown
    }

    override fun setDropDownViewResource(@LayoutRes resource: Int) {
        layoutResDropDownId = resource
        super.setDropDownViewResource(resource)
    }

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        var view = convertView
        // Get the data item for this position
        val taskList = getItem(position)!!
        // Check if an existing view is being reused, otherwise inflate the view
        val viewHolder: ViewHolder // view lookup cache stored in tag
        if (view == null) {
            view = LayoutInflater.from(context).inflate(layoutResId, parent, false)
            viewHolder = ViewHolder()
            viewHolder.name = view.findViewById(R.id.taskListTitle)
            view.tag = viewHolder
        } else {
            viewHolder = view.tag as ViewHolder
        }
        // Populate the data into the template view using the data object
        viewHolder.name.text = taskList.title
        // Return the completed view to render on screen
        return view!!
    }

    override fun getDropDownView(position: Int, convertView: View?, parent: ViewGroup): View {
        var view = convertView
        val taskList = getItem(position)!!
        val viewHolder: ViewHolder
        if (view == null) {
            view = LayoutInflater.from(context).inflate(layoutResDropDownId, parent, false)
            viewHolder = ViewHolder()
            viewHolder.name = view.findViewById(R.id.taskListTitle)
            view.tag = viewHolder
        } else {
            viewHolder = view.tag as ViewHolder
        }
        viewHolder.name.text = taskList.title
        return view!!
    }

    // View lookup cache
    internal class ViewHolder {
        lateinit var name: TextView
    }
}
